### ISSUE TITLE
Can't edit marameters (sometimes, undertermined)

### ISSUE THREAD

**Issue by edugimeno** [sugarlabs/turtleblocksjs#468]
Im using the online server.
Quite often, I can't tell when, suddenly the box parameters can't be edited. Like if I drag a "Rotate right" box to the screen with a parameter of "90" (degrees), quite often I will no longer be able to open and edit these 90 degrees box. When it happens, none of the parameters (forward X, backward X, left X...) would be editable again

Sometime, a full page refresh (F5) has helped, but some other times nothing has helped. 

At times it happens randomly, letting you working for 20 minutes, and then suddenly it stops working.

Everything else work, like I can still drag blocks, add new ones, run or save my project, but parameter editing is stuck.,

This is happening on Chrome, I tried on IE but the initial startup screen would never finish loading...

Thanks

**Comment by walterbender** [comment]
TurtleBlocksJS is a bit behind in its updates. Can you reproduce the same problem in Music Blocks?

**Comment by edugimeno** [comment]
> TurtleBlocksJS is a bit behind in its updates. Can you reproduce the same problem in Music Blocks?

Thanks, I realized after my comment that the Issues is completly outdated since like 2 years ago. I have been using this tool for my IT classes with mi kids so they learn some basic programming (and "sort-of" graphical programing) while they use a Scratch-like interface. This issue didn't happen last years but it suddenly started to happen last week when we got to this section

I have just tried music blocks myself, which I had never used before, and I see that it contains most of the turtle and workflow commands from the turtle plus the music statements, but I can't see how to share a project, so my pupils can send their work to me to be checked, as with Turtles JS I told them to use the Share Project (using download pulls a .html file that I can't find how to open to see the commands, just show the result)

I will keep trying, thanks for your reply!

**Comment by walterbender** [comment]
The Turtle code should be embedded in the HTML.
And sharing on the Planet should be working.

**Comment by edugimeno** [comment]
How do I access the code in the HTML? When I open an HTML page generated by "Save Project as HTML", all I get is like a preview of the generated shape and some fixed text, I can't see how to take a look at the code that generated that shape, what am I missing?

[LINK OF TEST HTML RESULT
](https://ibb.co/H4ZmTLX)
Thanks in advance!

**Comment by walterbender** [comment]
What happens when you click on the show button?

![Screenshot from 2021-04-12 11-26-25](https://user-images.githubusercontent.com/3996641/114420013-ef21fa80-9b81-11eb-8e0a-4ad14521d391.png)

**Comment by walterbender** [comment]
You should be able to copy/paste that code right into Music Blocks.

**Comment by walterbender** [comment]
or just drag and drop the HTML file into Music Blocks as well.

**Comment by edugimeno** [comment]
Upps ...wow... Well never expected this behaviour. Ok, if I drag the html code into a blank turtles screen, I do get the commands. I didn't know that this worked like this.
This should suffice to me for now for the pupils to properly turn their homework in
However if I copy the code under "show" (eg: [[0,["start",{"id":1618172972157,"collapsed":false,..........) into a blank turtles page, I get what looks like the correct blocks for the first time, but if U remove the blocks and paste them again, I get an "Could not parse JSON input". I don't really mind, first option (html dragging onto screen) works great for me.

But I still experience the inability to edit parameters from time to time. This does not happen at home but does happen quite often to my students while at school, I guess that going thru a cached and slow network connection could be related too.

Thanks a lot for your time!

**Comment by walterbender** [comment]
> But I still experience the inability to edit parameters from time to time. 

Probably a bug. What browser are they using?

**Comment by edugimeno** [comment]
Chrome. I also tried with IE but the initial colour changing turtle (splash screeen) never finished so we're stuck with Chrome

**Comment by walterbender** [comment]
IE was never supported but Edge should work (although I don't have any windows machines so it is not well tested.) But Chrome should work reasonably well. Next time it happens, it would great if you could ask the student to go to the console (Ctrl-Shift-J) and copy the log.

**Comment by edugimeno** [comment]
Ok I didn't know about that logging console functionality but I will keep it in mind tomorrow when I have class with them,
We don't have Edge as these computers are all using W7
I talked to the IT guy today and he said all the computers have Windows, Chrome and Java updates automatically pushed by policies so it should all be up to date
He explicitly added sugarlabs.org to the whitelist in the firewall so will see tomorrow.

I will keep you posted, many thanks again

**Comment by walterbender** [comment]
W7 suggests that the version of Chrome is prob. pretty old... and that version of IE is definitely not going to work :(

**Comment by edugimeno** [comment]
Ok, I tried yesterday and today in our class. Not very good. Log console did work on my computer, but for some reason, it's not working on the students computers, just not doing anything when I press Ctrl-Shift-J. Maybe related to different user role in Windows?
Also I wanted to see if it makes any difference under "incognito- private browsing" under Chrome but for unknow reasons this option was not in these Chrome browsers, again maybe windows role related (regular user?)
ALso like last time, turtle blocks were randomly stopping responding to parameter editting attemps, and sometimes randomly getting back to doing it right again. We tried Music Blocks but same issue was experienced on this App too

**Comment by walterbender** [comment]
Thanks for trying. Sounds like you'd need the IT team to help with the Windows/Chrome issues. But maybe if the problem occurs on your computer you could grab the log for me? (and maybe a screenshot?)

**Comment by edugimeno** [comment]
Yes I can try, it did happen on my "teacher" computer once, when I was projecting my screen to my students as an introductory class for this section, but I didn't know about the logging funcion at that time. It didn't happen again on mine.
About screeenshot I can take it but it won't help as you will only see a normal screen with my cursor no the parameter block. I can take a short video, basically the effect is that when for example you drag a "right 90" block to the screen, and then click into the 90 so you can change it to 180, you should get a pop-up window where you can either edit the value, or use the up/down controls to modify it. In this issue no matter how many times I click/double click on the "90", nothing changes and there's no way to edit this value. Same with other editable numeric/variables boxes from that moment on.

**Comment by walterbender** [comment]
I will try to reproduce the error locally.
